examples/name-service-without-macros/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,219 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::bytes::AsByteVec;
+use light_sdk::{
+    compressed_account::{LightAccount, LightAccounts, OutputCompressedAccountWithPackedContext},
+    context::LightInstructionInputs,
+    light_system_accounts, LightDiscriminator, LightHasher, LightTraits,
+};
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+    use light_sdk::context::LightContext;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateRecord<'info>>,
+        inputs: Vec<u8>,
+    ) -> Result<()> {
+        let inputs = CreateRecordInputs::deserialize(&mut inputs.as_slice())?;
+        let mut ctx =
+            LightContext::<CreateRecord, LightCreateRecord>::new(ctx, inputs.light_inputs)?;
```

Comment:
how does this handle multiple compressed accounts?

Reply:
In the manual example, you would define your `Light*` struct as following:

```rust
pub struct LightUpdateRecord {
    pub record: LightAccount<NameRecord>,
    // a new cPDA
    pub record_history: LightAccount<RecordHistory>,
}
```

And implement the `LightAccounts` trait as following:

```rust

impl LightAccounts for LightUpdateRecord {
    fn try_light_accounts(inputs: &LightInstructionInputs) -> Result<Self> {
        let record: LightAccount<NameRecord> = LightAccount::try_from_slice_mut(
            &inputs.accounts.as_ref().unwrap()[0].compressed_account,
        )?;
        let record_history: LightAccount<RecordHistory> = LightAccount::try_from_slice_mut(
            &inputs.accounts.as_ref().unwrap()[1].compressed_account,
        )?;
        Ok(Self { record, record_history })
    }

    fn output_accounts(&self) -> Result<Vec<OutputCompressedAccountWithPackedContext>> {
        let mut output_accounts = Vec::with_capacity(2);
        if let Some(record_output_account) = self.record.output_compressed_account()? {
            output_accounts.push(record_output_account);
        }
        if let Some(record_history_output_account) = self.record_history.output_compressed_account()? {
            output_accounts.push(record_history_output_account);
        }
        Ok(output_accounts)
    }
}
```

I still need to clean that up from unwraps - I think I will add our own error type. The current implementation on main doesn't have unwraps, they are added in this branch to handle the `data: Option`.

Macros are generating such code automatically based on your `#[light_account]` fields in the struct annotated with `#[light_accounts]`. The intermediate step which macros take is moving `#[light_account]` fields into a separate struct with `Light` prefix.